src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs

```diff
@@ -111,11 +120,139 @@ public async Task<ActionResult> Add(AddAatfViewModel viewModel)
             }
         }
 
+        [HttpGet]
+        public ActionResult Type(string searchedText)
+        {
+            return View(new OrganisationTypeViewModel(searchedText));
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Type(OrganisationTypeViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var organisationType = model.SelectedValue.GetValueFromDisplayName<OrganisationType>();
+
+                switch (organisationType)
+                {
+                    case OrganisationType.SoleTraderOrIndividual:
+                    case OrganisationType.Partnership:
+                        return RedirectToAction("SoleTraderOrPartnershipDetails", "AddAatf", new { organisationType = model.SelectedValue, searchedText = model.SearchedText });
+                    case OrganisationType.RegisteredCompany:
+                        return RedirectToAction("RegisteredCompanyDetails", "AddAatf", new { organisationType = model.SelectedValue, searchedText = model.SearchedText });
+                }
+            }
+
+            return View(model);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> SoleTraderOrPartnershipDetails(string organisationType, string searchedText = null)
+        {
+            IList<CountryData> countries = await GetCountries();
+
+            SoleTraderOrPartnershipDetailsViewModel model = new SoleTraderOrPartnershipDetailsViewModel
+            {
+                BusinessTradingName = searchedText,
+                OrganisationType = organisationType
+            };
+
+            model.Address.Countries = countries;
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> SoleTraderOrPartnershipDetails(SoleTraderOrPartnershipDetailsViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                IList<CountryData> countries = await GetCountries();
+
+                model.Address.Countries = countries;
+                return View(model);
+            }
+
+            using (var client = apiClient())
+            {
+                CreateOrganisationAdmin request = new CreateOrganisationAdmin()
```

Comment:
Add a unit test for this... when model is valid a request is created with valid parameters

Can we also add a similar unit test for the Add AATF post method so the AddAtf request is called with correct parameters

Reply:
Done